src/api/process.c

```diff
@@ -560,12 +560,12 @@ static int process_env_add(process_env_t *env_list, size_t *env_len, const char
 }
 
 
-static void process_env_free(process_env_t *list) {
+static void process_env_free(process_env_t *list, size_t list_len) {
   if (!*list) return;
 #ifdef _WIN32
   free(*list);
 #else
-  for (size_t i = 0; (*list)[i]; i++) free((*list)[i]);
+  for (size_t i = 0; i < list_len; i++) free((*list)[i]);
```

Comment:
Can you move the `for` loop into its own ifdef (`#ifndef _WIN32`) and then the free() outside? Something like:

```
#ifndef _WIN32
for () blabla();
#endif
free(*list);
```